.github/workflows/azure-podvm-image-nightly-build.yml

```diff
@@ -0,0 +1,11 @@
+name: azure podvm image nightly build
+
+on:
+  schedule:
+    # Run at 12:00 AM UTC
+    - cron: '0 0 * * *'
+
+jobs:
+  build-podvm-image:
+    uses: confidential-containers/cloud-api-adaptor/.github/workflows/azure-podvm-image-build.yml@main
+    secrets: inherit
```

Comment:
For the nightly we need to pass the image version as: `yyyy.mm.dd` format. As done here: https://github.com/confidential-containers/cloud-api-adaptor/pull/1338/files#diff-5105e3199531bd6cd2f41cad8b6fdeff75775347857c7286e8af895825690d78R24-R29

```yaml
    - name: Generate version for pod vm image
      id: generate_image_version
      run: |
        unique_version=$(date +'%Y.%m.%d')
        echo "Generated unique version for the image as: ${unique_version}"
        echo "pod_image_version=${unique_version}" >> $GITHUB_OUTPUT
```

Reply:
Ah, I thought both nightly and e2e have the same versioning theme. 

In this case we can just make `image-version` a mandatory input in the build workflow and generate it on the caller side (2e-test or nightly-build). That's a bit cleaner, actually.